src/v2/Apps/Artist/ArtistApp.tsx

```diff
@@ -45,8 +45,26 @@ const ArtistApp: React.FC<ArtistAppProps> = ({ artist, children, match }) => {
   }
 
   const showOverviewTab = hasOverviewContent(artist)
-  const showArtworksTab = artist?.statuses?.artworks
-  const showAuctionResultsTab = artist?.statuses?.auctionLots
+  const showAuctionResultsTab =
+    artist?.statuses?.auctionLots || artist?.auctionResultsStatus?.edges?.length
+  const hasArtworks = artist?.statuses?.artworks
+
+  let showArtworksTab
+  if (hasArtworks) {
+    showArtworksTab = true
+
+    // We use a view of the artworks tab zero state even if there's no artworks
+    // and hide the tab. However, often an artist has no artworks but a lot of
+    // past auction results, and in that case, we show the artworks tab because
+    // we need to also show the auction results tab.
+  } else if (!showOverviewTab && showAuctionResultsTab) {
+    showArtworksTab = true
+
+    // By default, if there's no content, show artworks page with zerostate but
+    // also hide the tab.
+  } else if (!showOverviewTab && !showArtworksTab && !showAuctionResultsTab) {
+    showArtworksTab = false
+  }
```

Comment:
`non-blocking`: I find assigning booleans like this kinda weird. Presumably this is equivalent to:

```js
const showArtworksTab =
  hasArtworks ||
  // We use a view of the artworks tab zero state even if there's no artworks
  // and hide the tab. However, often an artist has no artworks but a lot of
  // past auction results, and in that case, we show the artworks tab because
  // we need to also show the auction results tab.
  (!showOverviewTab && showAuctionResultsTab) ||
  // By default, if there's no content, show artworks page with zerostate but
  // also hide the tab.
  !(!showOverviewTab && !showArtworksTab && !showAuctionResultsTab);
```

But if the former is more clear; by all means.

Reply:
Going with the former 😄. Verbose and old school, if ugly